### ISSUE TITLE
Upgrade to Kubernetes 1.8

### ISSUE THREAD

**Issue by asymmetric** [kinvolk/habitat-operator#102]
* Update `client-go` to 1.8
* Update README

See [this](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG.md/#introduction-to-v180) for changes.

**Comment by LiliC** [comment]
I tested the operator with `1.8.0` and all the resources are created correctly and the desired functionally is working. Not sure what else you meant by `Upgrade to Kubernetes 1.8`? Maybe we can add in details which versions we support (have been tested) in the `README`, but other then that not sure if there is anything else we need to do here.

**Comment by asymmetric** [comment]
Sorry for not having been clearer. I guess the most important point is upgrading to the version of client-go meant for 1.8. One hasn't come out yet though.

Also, since we're talking about it, I wonder if we can actually say we support '>= 1.7'. Meaning, that at the moment we only test against 1.7, and that's it.

We should figure out what versions we can/want to support, considering that the `client-go` library is tied to the Kubernetes version.

**Comment by LiliC** [comment]
> Also, since we're talking about it, I wonder if we can actually say we support '>= 1.7'. Meaning, that at the moment we only test against 1.7, and that's it.

Yes, good point, I was already thinking of this. Was thinking if maybe we can extend our tests and test against all of `1.7.x` that are available. Maybe after merging to master we could have that test. Or something similar. That would be one of ways to ensure we always support the versions.

Can you update the description with more details, that this involves upgrading the `client-go` library etc. Maybe we can also move it to blocked as it's blocked on the release of  `client-go``1.8`. Thanks.

**Comment by asymmetric** [comment]
I don't know if adding more tests (I suppose you meant integration tests) is the right approach, given how long they take to run already.

Let's see once we have a newer `client-go` version.

**Comment by LiliC** [comment]
Closing in favour of https://github.com/kinvolk/habitat-operator/issues/137.
